gwsurrogate/new/precessing_surrogate.py

```diff
@@ -152,64 +151,33 @@ def transformTimeDependentVector(quat, vec):
 
 ###############################################################################
 # Functions related to fit evaluations
+def _eval_scalar_fit(fit_data, fit_params, get_fit_settings):
+    """ Evaluates a single scalar fit.
 
-def _get_fit_settings():
-    """
-     These are to rescale the mass ratio fit range
-     from [-0.01, np.log(4+0.01)] to [-1, 1]. The chi fits are already in
-     this range.
-
-
-     Values defined here are model-specific. These values are for NRSur7dq4.
-    """
-    q_fit_offset = -0.9857019407834238
-    q_fit_slope = 1.4298059216576398
-    q_max_bfOrder = 3
-    chi_max_bfOrder = 2
-    return q_fit_offset, q_fit_slope, q_max_bfOrder, chi_max_bfOrder
-
-def _get_fit_params(x):
-    """ Converts from x=[q, chi1x, chi1y, chi1z, chi2x, chi2y, chi2z]
-        to x = [np.log(q), chi1x, chi1y, chiHat, chi2x, chi2y, chi_a]
-        chiHat is defined in Eq.(3) of 1508.07253.
-        chi_a = (chi1 - chi2)/2.
-        Both chiHat and chi_a always lie in range [-1, 1].
-    """
-
-    x = np.copy(x)
-
-    q = float(x[0])
-    chi1z = float(x[3])
-    chi2z = float(x[6])
-    eta = q/(1.+q)**2
-    chi_wtAvg = (q*chi1z+chi2z)/(1+q)
-    chiHat = (chi_wtAvg - 38.*eta/113.*(chi1z + chi2z)) \
-        /(1. - 76.*eta/113.)
-    chi_a = (chi1z - chi2z)/2.
-
-    x[0] = np.log(q)
-    x[3] = chiHat
-    x[6] = chi_a
-
-    return x
+        fit_params should come from each surrogate model's
+        self._get_fit_params()
 
-def _eval_scalar_fit(fit_data, fit_params):
-    """ Evaluates a single scalar fit.
-        fit_params should come from _get_fit_params()
+        get_fit_settings should come from each surrogate model's
+        self._get_fit_settings
     """
     q_fit_offset, q_fit_slope, q_max_bfOrder, chi_max_bfOrder \
-        = _get_fit_settings()
+        = get_fit_settings()
     val = _utils.eval_fit(fit_data['bfOrders'], fit_data['coefs'], \
         fit_params, q_fit_offset, q_fit_slope, q_max_bfOrder, chi_max_bfOrder)
     return val
 
-def _eval_vector_fit(fit_data, size, fit_params):
+def _eval_vector_fit(fit_data, size, fit_params, get_fit_settings):
     """ Evaluates a vector fit, where each element is a scalar fit.
-        fit_params should come from _get_fit_params()
+
+        fit_params should come from each surrogate model's
+        self._get_fit_params()
+
+        get_fit_settings should come from each surrogate model's
+        self._get_fit_settings
```

Comment:
This documentation doesn't seem to match up with the code

Reply:
Comments now clarify what the code actually does and where the data/methods are set.